### PR TITLE
Update pigweed to 78d0730d

### LINKED ISSUE #7565: vscode bootstrap failing with EXDEV
```
Downloading and installing packages into local source directory:
  Setting up CIPD package manager...done
  Setting up Python environment.....[\]
ninja: Entering directory `/home/vscode/pigweed/env/gn_out'
[1/141] touch protocol_buffer/obj/third_party/pigweed/repo/pw_unit_test/unit_test_proto._includes.stamp
[2/141] touch protocol_buffer/obj/third_party/pigweed/repo/pw_rpc/protos._includes.stamp
[3/141] touch python/obj/third_party/pigweed/repo/pw_build/python_lint._run_pip_install.stamp
[4/141] python3 ../../../../../workspaces/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root /workspaces/connectedhomeip/ --current-path /workspaces/connectedhomeip/third_party/happy --default-toolchain=//build/toolchain/host:linux_x64_gcc --current-toolchain=//third_party/pigweed/repo/pw_build/python_toolchain:python --touch /home/vscode/pigweed/env/gn_out/python/gen/third_party/happy/happy.pw_pystamp --capture-output -- /workspaces/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/nop.py
[5/141] python3 ../../../../../workspaces/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root /workspaces/connectedhomeip/ --current-path /workspaces/connectedhomeip/third_party/pigweed/repo/pw_cli/py --default-toolchain=//build/toolchain/host:linux_x64_gcc --current-toolchain=//third_party/pigweed/repo/pw_build/python_toolchain:python --touch /home/vscode/pigweed/env/gn_out/python/gen/third_party/pigweed/repo/pw_cli/py/py.pw_pystamp --capture-output -- /workspaces/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/nop.py
[6/141] touch python/obj/third_party/happy/happy.stamp
[7/141] python3 ../../../../../workspaces/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root /workspaces/connectedhomeip/ --current-path /workspaces/connectedhomeip/integrations/mobly --default-toolchain=//build/toolchain/host:linux_x64_gcc --current-toolchain=//third_party/pigweed/repo/pw_build/python_toolchain:python --touch /home/vscode/pigweed/env/gn_out/python/gen/integrations/mobly/chip_mobly.pw_pystamp --capture-output -- /workspaces/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/nop.py
[8/141] touch python/obj/third_party/pigweed/repo/pw_cli/py/py.stamp
[9/141] touch python/obj/integrations/mobly/chip_mobly.stamp
[10/141] python3 ../../../../../workspaces/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root /workspaces/connectedhomeip/ --current-path /workspaces/connectedhomeip/third_party/pigweed/repo/pw_unit_test --default-toolchain=//build/toolchain/host:linux_x64_gcc --current-toolchain=//third_party/pigweed/repo/pw_protobuf_compiler/toolchain:protocol_buffer --capture-output -- /workspaces/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/mirror_tree.py --source-root /workspaces/connectedhomeip/third_party/pigweed/repo/pw_unit_test/ --directory /home/vscode/pigweed/env/gn_out/protocol_buffer/gen/third_party/pigweed/repo/pw_unit_test/unit_test_proto.proto_library/sources/ /workspaces/connectedhomeip/third_party/pigweed/repo/pw_unit_test/pw_unit_test_proto/unit_test.proto
FAILED: protocol_buffer/gen/third_party/pigweed/repo/pw_unit_test/unit_test_proto.proto_library/sources/pw_unit_test_proto/unit_test.proto 
python3 ../../../../../workspaces/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root /workspaces/connectedhomeip/ --current-path /workspaces/connectedhomeip/third_party/pigweed/repo/pw_unit_test --default-toolchain=//build/toolchain/host:linux_x64_gcc --current-toolchain=//third_party/pigweed/repo/pw_protobuf_compiler/toolchain:protocol_buffer --capture-output -- /workspaces/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/mirror_tree.py --source-root /workspaces/connectedhomeip/third_party/pigweed/repo/pw_unit_test/ --directory /home/vscode/pigweed/env/gn_out/protocol_buffer/gen/third_party/pigweed/repo/pw_unit_test/unit_test_proto.proto_library/sources/ /workspaces/connectedhomeip/third_party/pigweed/repo/pw_unit_test/pw_unit_test_proto/unit_test.proto
Traceback (most recent call last):
  File "/workspaces/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/mirror_tree.py", line 116, in <module>
    mirror_paths(**vars(_parse_args()))
  File "/workspaces/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/mirror_tree.py", line 106, in mirror_paths
    outputs = list(_link_files(source_root, sources, directory))
  File "/workspaces/connectedhomeip/third_party/pigweed/repo/pw_build/py/pw_build/mirror_tree.py", line 60, in _link_files
    os.link(source, dest)
OSError: [Errno 18] Invalid cross-device link: 
```

### PR DESCRIPTION
**Pull request by mspang**
### Problem

Bootstrapping in VS Code results in EXDEV.

Pigweed is out of date.

### Summary of changes

Update https://github.com/google/pigweed/compare/4afe7a4158fae3d56f8282c0c114880f407ab105...626a9b8e988ef9f1bb1c5bc809a3490f96caf86c

Add dependencies to executables to satisfy the new scheme for pigweed assertion & logging.

### Testing

Bootstrap and check pyparsing version in `pip list`.

Bootstrap in vs code and confirm no error.

fixes #7565, #7386

### CODE REVIEW

**Review by woody-apple — DISMISSED** [review]
@mspang Can you fill out the template? How was this tested? What new features are we getting for updating?

File: scripts/environment.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
-    "cipd_package_files": ["scripts/pigweed.json"],
+    "cipd_package_files": [
+        "third_party/pigweed/repo/pw_env_setup/py/pw_env_setup/cipd_setup/pigweed.json",
```
Comment (Damian-Nordic) [inline scripts/environment.json:3]: The default `pigweed.json` contains some packages which we don't need like `go`? Removing those saved up to 20s per CI build as far as I remember. Do you think it's not worth maintaining it anyway?

**Review by andy31415 — APPROVED** [review]
Approving pending summary update (for testing and short TLDR so that we do not need to click on fixes links)

**Review by msandstedt — APPROVED** [review]

File: scripts/environment.json (hunk 1 above)
Comment (mspang) [reply scripts/environment.json:3]: Pigweed split out cpython for us, they might be willing to split it up even further.

I don't mind removing go, although I think it's possible we'll end up wanting to use some of the go based tools eventually.

The dumb thing is that GitHub re-bootstraps every time; we don't have that problem in our infrastructure so the cost of carrying these is amortized down to nothing.

**Review by chrisdecenzo — APPROVED** [review]
I am only able to verify the pigweed repo update (which fixed the problems I was having)

**Pull request merged**